### PR TITLE
network error, initialize error 

### PR DESCRIPTION
**Pull request by ice-orion**

### CODE REVIEW

File: src/utils/errors.tsx [hunk 1]
```diff
@@ -10,30 +10,36 @@ import {checkProp} from '@utils/guards';
...
-      message: t('pop_up.try_again_text', {
-        errorIdentifier: `TODO set error identifier: ${errorIdentifier}`,
-      }),
+      title: isNetworkError(error)
+        ? t('errors.network_error_title')
+        : t('errors.general_error_title'),
+      message: isNetworkError(error)
+        ? getErrorMessage(error)
+        : t('errors.general_error_message', {
+            errorMessage: getErrorMessage(error),
```
Comment (ice-ares) [inline src/utils/errors.tsx:39]: don't show it

Comment (ice-orion) [reply src/utils/errors.tsx:39]: No? Why? I thought this is the main goal - to show the error message from the BE that contains error status/code/error identifier and so on?

File: src/translations/locales/en.json [hunk 2]
```diff
@@ -170,7 +172,13 @@
...
     "max_username": "Maximum length is 30 characters.",
     "code_expired": "The code has expired. Please ask for a new code.",
-    "too_many_requests": "Too many requests. Please try again later."
+    "too_many_requests": "Too many requests. Please try again later.",
+    "network_error_title": "No internet connection",
+    "network_error_message": "Your device does not have any active internet connection",
+    "unknown_error": "Some error occurred, please try again later",
+    "api_error": "Request failed with status {{status}} and code {{code}}",
+    "general_error_title": "Oups...",
+    "general_error_message": "Looks like something went wrong. Our team is working on fixing it right now\n{{errorMessage}}"
```
Comment (ice-hector) [inline src/translations/locales/en.json:181]: Lets remove \n and {{errorMessage}} from here and do \n in code and then display the ```errorMessage``` itself

File: src/components/InitializationError/index.tsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        style={styles.image}
+        source={Images.popUp.error}
+      />
+      <Text style={styles.titleText}>{t('errors.network_error_title')}</Text>
+      <Text style={styles.messageText}>
+        {t('errors.network_error_message')}
+      </Text>
+      <PrimaryButton
+        style={styles.button}
+        onPress={() => RNRestart.Restart()}
```
Comment (ice-hector) [inline src/components/InitializationError/index.tsx:27]: First of all, let's move to separate const function
And does it kill the app? Or if not: how will such reload look in prod? 
And also as I see from their [docs](https://github.com/avishayil/react-native-restart#usage) you are using the deprecated `RNRestart.Restart()` method

File: src/navigation/Main.tsx [hunk 4]
```diff
@@ -261,6 +262,16 @@ const MainTabs = () => {
...
+  useEffect(navigationReadyResolver, []);
+
+  // TODO: Hide until functionality is ready
+  // Warning! Calling this hook with no internet leads to the app hanging (setTimeout, button handles, requests don't work)
+  // So:
+  //  1. The problem should be investigated
+  //  2. Should it be called only for authenticated users?
+  //    If so, then it should be called here instead of Router.tsx or even better, via sagas
+  //    + calling in Router.tsx leads to splash screen hanging if user doesn't have internet connection during the app opening
+  // useGetstreamListener();
```
Comment (ice-hector) [inline src/navigation/Main.tsx:274]: Did we remove it totally? Or it's just commented until we find the problem that causes such behavior when we setup getstream?

Comment (ice-orion) [reply src/navigation/Main.tsx:274]: As i see the whole notification's usage is commented with `TODO: Hide until functionality is ready` comment, so this is also disabled until "functionality is ready" + the problem is found and solved

File: src/components/InitializationError/index.tsx (hunk 3 above)
Comment (ice-orion) [reply src/components/InitializationError/index.tsx:27]: You can look into the sources to find out how it works (on iOS RCTTriggerReloadCommandListeners, on android it uses https://github.com/JakeWharton/ProcessPhoenix). 

>how will such reload look in prod?
You can download the build from firebase, torn off the internet and try to launch the app - from the user's perspective the app just reloads 

> And also as I see from their [docs](https://github.com/avishayil/react-native-restart#usage) you are using the deprecated RNRestart.Restart() method
updated the lib, `RNRestart.restart` is now used (exactly the same as `RNRestart.Restart` - they just renamed the function 🙈)

File: src/components/InitializationError/index.tsx [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+import {t} from '@translations/i18n';
+import {font} from '@utils/styles';
+import React from 'react';
+import {Image, StyleSheet, Text, View} from 'react-native';
+import RNRestart from 'react-native-restart';
+import {rem} from 'rn-units';
+
+export const InitializationError = () => {
+  const onTryAgainPress = () => {
+    RNRestart.restart();
```
Comment (ice-ares) [inline src/components/InitializationError/index.tsx:15]: U're restarting the app cuz of no network connection?

File: src/utils/errors.tsx (hunk 1 above)
Comment (ice-orion) [reply src/utils/errors.tsx:39]: done

File: src/translations/locales/en.json (hunk 2 above)
Comment (ice-orion) [reply src/translations/locales/en.json:181]: done, we don't need `errorMessage` now at all

File: src/components/InitializationError/index.tsx (hunk 5 above)
Comment (ice-orion) [reply src/components/InitializationError/index.tsx:15]: This is the screen / component that is displayed to the user if app didn't manage to initialize. And the button "try again" - to user can press it to try to launch the app once again (basically restart)

**Review by ice-hector — APPROVED** [review]

**Review by ice-ares — APPROVED** [review]

**Pull request merged**